File: Session4/C++/ValidAnagram.cpp
https://leetcode.com/problems/valid-anagram/

class Solution {
public:
    bool isAnagram(string s, string t) {
        
        if(s.size()!=t.size()){
            return false;
        }
        
        vector<int> map(26, 0);
        
        for(int i=0; i<s.size(); i++) {
            map[s[i]-'a']++;
            map[t[i]-'a']--;
        }
        
        for(int i=0; i<26; i++) {
            if(map[i]!=0) {
                return false;
            }
        }
        
        return true;
    }
};

File: Session4/Java/ValidAnagram.java
https://leetcode.com/problems/valid-anagram/

class Solution {
    public boolean isAnagram(String s, String t) {
        
        if(s.length()!=t.length()) {
            return false;
        }
        
        int[] map = new int[26];
        
        for(int i=0; i<s.length(); i++) {
            map[s.charAt(i)-'a']++; // inc
            map[t.charAt(i)-'a']--; // dec
        }
        
        for(int i:map) {
            if(i!=0) {
                return false;
            }
        }
        
        return true;
    }
}